File: Assets/Scripts/MainScript.js
#pragma strict


static var EditMode:boolean;
static var Selected:GameObject;


static var yellow:Material;
static var red:Material;
static var blue:Material;
 
 
static var curLevel:String;
var nonStaticCurLevel:String;
static var lastLevel:String;
static var lastExit:String;

 
function Start()
{
	curLevel=nonStaticCurLevel;
 }
 
function Update () 
{
	if (!EditMode) 
		return;     
    
    var mousex = Input.mousePosition.x;
    var mousey = Input.mousePosition.y;
    var ray = GetComponent.<Camera>().main.ScreenPointToRay (Vector3(mousex,mousey,0));
          
    if ( ( Input.GetMouseButtonDown(0)) &&
    (Input.GetKey(KeyCode.RightControl) || Input.GetKey(KeyCode.LeftControl)))
    {
	    var crate = Instantiate(Prefabs[enemyIndex], ray.origin, Quaternion.identity);
	    crate.transform.position.z=0;   
	    Selected=crate;
    }
    
    if (Input.GetKeyUp("space"))
    	enemyIndex++;
    
    if (enemyIndex>=Prefabs.length)
    	enemyIndex=0;    	
}



var Prefabs:GameObject[];
static var enemyIndex:int=0;
static var verticalCam:boolean=false;
function OnGUI()
{

 GUI.depth = 0;
	GUI.Label(Rect(0,100,100,100),"Scene"+curLevel);
	
	//test door output	
	if(false)	
	for (var j:int;j<70;j++)
	{
		GUI.Label(Rect(j*50,50,100,100),j+""+ElevatorCodes.TVCleared(j));	
	}
	
	
	GUI.Label(Rect(0,100,100,100),"Scene"+curLevel);
	
	if (!EditMode)
		return;
	
	GUI.Label(Rect(0,0,200,100),"CREATE: "+Prefabs[enemyIndex].transform.name);
	
	var SelectedName:String ="";
	if (Selected!=null)
		SelectedName =Selected.transform.name;

	GUI.Label(Rect(0,20,200,100),"Selected: "+SelectedName);
	if (GUI.Button(Rect(0,40,200,20),"Vertical Cam: "+verticalCam))
		verticalCam = !verticalCam;
}

File: Assets/Scripts/MainScript.cs
using UnityEngine;

[System.Serializable]
public partial class MainScript : MonoBehaviour
{
    public static string GameType; //hacked for Game B to play list of scenes
    public static string[] SceneList; //hacked for Game B to play list of scenes
    public static int SceneListNumber; //hacked for Game B to play list of scenes

    public static bool EditMode;
    public static GameObject Selected;
    public static Material yellow;
    public static Material red;
    public static Material blue;
    public static string curLevel;
    public string nonStaticCurLevel;
    public static string lastLevel = "Scene101";
    public static string lastExit;

    public static float timer = -100;
    public virtual void Start()
    {
        curLevel = nonStaticCurLevel;
    }

    public virtual void Update()
    {
        if (!MainScript.EditMode)
        {
            return;
        }
        float mouseX = Input.mousePosition.x;
        float mouseY = Input.mousePosition.y;
        Ray ray = Camera.main.ScreenPointToRay(new Vector3(mouseX, mouseY, 0));
        if (Input.GetMouseButtonDown(0) 
        && (Input.GetKey(KeyCode.RightControl) || Input.GetKey(KeyCode.LeftControl)))
        {
            GameObject crate = UnityEngine.Object.Instantiate(Prefabs[MainScript.enemyIndex], ray.origin, Quaternion.identity);

            //set crate z position to 0
            crate.transform.position = new Vector3(crate.transform.position.x, crate.transform.position.y,0);
            MainScript.Selected = crate;
        }
        if (Input.GetKeyUp("space"))
            MainScript.enemyIndex++;
        if (MainScript.enemyIndex >= Prefabs.Length)
            MainScript.enemyIndex = 0;
    }

    public GameObject[] Prefabs;
    public static int enemyIndex;
    public static bool verticalCam;
    public virtual void OnGUI()
    {
        GUI.depth = 0;
        GUI.Label(new Rect(0, 100, 100, 100), "Scene" + MainScript.curLevel);
        GUI.Label(new Rect(0, 100, 100, 100), "Scene" + MainScript.curLevel);
        if (!MainScript.EditMode)
            return;

        GUI.Label(new Rect(0, 0, 200, 100), "CREATE: " + Prefabs[MainScript.enemyIndex].transform.name);
        string SelectedName = "";
        if (MainScript.Selected != null)
            SelectedName = MainScript.Selected.transform.name;
    
        GUI.Label(new Rect(0, 20, 200, 100), "Selected: " + SelectedName);
        if (GUI.Button(new Rect(0, 40, 200, 20), "Vertical Cam: " + MainScript.verticalCam))
            MainScript.verticalCam = !MainScript.verticalCam;
    }
}